[IMP] runbot: add troubleshooting documentation (#750)
odoo/runbot: runbot/documentation/troubleshooting.md

# Troubleshooting

When a build fails or a ci is missing it can be for multiple reasons, and runbot offers multiple features to fix this depending on the issue. Note that most actions are only available to logged in users. 

## Check batch informations

> The batch view can offer precious informations, like not synchronous commit (the different branches where likely not rebased at the same time, on compatible commits) or the commits are maybe to old. In this case the first action should be to rebase all branches by ensuring to fetch all repo in the same time frame.

Reviewer: ```suggestion
The batch view can offer precious informations, like not synchronous commit (the different branches were likely not rebased at the same time, on compatible commits) or the commits are maybe to old. In this case the first action should be to rebase all branches by ensuring to fetch all repo in the same time frame.
```

![warning](images/warning1.png)
![warning](images/behind.png)

## Red build

A build can be red for an unrelated random error or time related errors. In this case the build can be replayed with a rebuild. It is also usefull to apply some changes made in the runbot backend (configuration changes, build errors disabling a specific tests, upgrade exception, ...)

Most of the time, in this case, a rebuild should be enough

![warning](images/rebuild_main.png)

Note that for build with children, it is usually enough to rebuild only the red child. This is faster and creates less load.

![warning](images/rebuild_child.png)

## Missing ci on github/other bot.

In some cases a signal between runbot and github, or github and mergebot can be lost on the path (network or github issues most of the time). In this case, a build can be green but the corresponding ci on github or on the mergebot dashboard can be missing.

> There is a action to resend the last commit status of a commit for this purpose.

Reviewer: ```suggestion
There is an action to resend the last commit status of a commit for this purpose.
```

> First access the commit page. Most of the commit sha displayed in the interface are clickable and will lead to the commit page.

Reviewer: ```suggestion
First access the commit page. Most of the commit sha1 hashes displayed in the interface are clickable and will lead to the commit page.
```
